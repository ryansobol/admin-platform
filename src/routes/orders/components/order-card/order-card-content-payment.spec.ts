import { render, screen } from '@testing-library/svelte';
import { expect, it } from 'vitest';

import OrderCardContentPayment from './order-card-content-payment.svelte';
import type { Order } from '../../types';

it('renders nothing without an order prop', () => {
	render(OrderCardContentPayment);

	const element = screen.queryByRole('region', { name: 'Order Payment Information' });

	expect(element).toBeNull();
});

it('renders the component with an order prop', () => {
	const order: Order = {
		id: 1,
		code: 'ABC123',
		customer: {
			name: 'Firstname Lastname',
			email: 'firstname.lastname@example.com',
			phone: '+1-234-567-8900'
		},
		type: 'Monthly Subscription',
		status: 'Fulfilled',
		createdAt: '2023-01-09T12:11:27-07:00',
		updatedAt: '2023-01-11T04:11:27-07:00',
		products: [{ id: 7, name: 'Handmade Ceramic Mug', quantity: 4, price: 304 }],
		subtotal: 1216,
		shipping: 26.99,
		tax: 109.44,
		total: 1352.43,
		shippingAddress: {
			line1: 'Firstname Lastname',
			line2: '117 1st Ave.',
			line3: 'Phoenix, NY 60975'
		},
		billingAddress: {
			line1: 'Firstname Lastname',
			line2: '117 1st Ave.',
			line3: 'Phoenix, NY 60975'
		},
		creditCard: {
			type: 'Visa',
			numberRedacted: '**** **** **** 3088'
		}
	};

	render(OrderCardContentPayment, { order });

	const element = screen.getByRole('region', { name: 'Order Payment Information' });

	expect(element).toMatchSnapshot();
});
