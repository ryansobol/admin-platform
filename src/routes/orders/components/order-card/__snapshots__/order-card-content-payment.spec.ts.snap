// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the component with an order prop 1`] = `
<section
  aria-label="Order Payment"
  class="grid gap-3"
>
  <h4
    class="font-semibold"
  >
    Payment
  </h4>
   
  <dl
    class="grid gap-3"
  >
    <div
      class="flex items-center justify-between"
    >
      <dt
        class="flex items-center gap-1 text-muted-foreground"
      >
        <svg
          class="lucide-icon lucide lucide-credit-card h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="14"
            rx="2"
            width="20"
            x="2"
            y="5"
          >
            
          </rect>
          <!---->
          
          <line
            x1="2"
            x2="22"
            y1="10"
            y2="10"
          >
            
          </line>
          <!---->
          
          <!---->
          <!---->
          
          <!---->
        </svg>
        <!---->
         Visa
      </dt>
       
      <dd>
        **** **** **** 3088
      </dd>
    </div>
  </dl>
</section>
`;
